tests/dialects/test_memref.py

```diff
@@ -114,3 +118,65 @@ def test_memref_dealloca():
     dealloc0 = Dealloca.get(alloc0)
 
     assert type(dealloc0.memref) is OpResult
+
+
+def test_memref_dim():
+    idx = arith.Constant.from_int_and_width(1, IndexType())
+    alloc0 = Alloc.get(i32, 64, [3, 1, 2])
+    dim_1 = memref.Dim.from_source_and_index(alloc0, idx)
+
+    assert dim_1.source is alloc0.memref
+    assert dim_1.index is idx.result
+    assert isinstance(dim_1.result.typ, IndexType)
+
+
+def test_memref_rank():
+    alloc0 = Alloc.get(i32, 64, [3, 1, 2])
+    dim_1 = memref.Rank.from_memref(alloc0)
+
+    assert dim_1.source is alloc0.memref
+    assert isinstance(dim_1.rank.typ, IndexType)
+
+
+def test_memref_matmul_verify():
```

Comment:
I personally disagree with having this as our tests. I feel it does not convey actual useful information to hunt bugs with but is only a big use case. Maybe I am missing something though.

Reply:
I personally think that we should make a pass on all of our tests, to clean these tests, and move them to "integration tests" or examples.
But right now, in our repo, we do have tests like this.
I'm quite busy in the next week, but maybe after I can think of the test dialect?

In the meantime, I would say that it is fine to have this here for now, since we need to clean up the entire repo anyway.
Though maybe you feel strongly about the contrary?

Reply:
Merging as is is okay with me, though I am not sure when we will get to actually fixing this (like the other about 2000 issues we have discussed before :wink: )